### PR TITLE
Rename `--mender-professional` setup flag to `--hosted-mender`

### PR DESCRIPTION
**Pull request by alfrunes**
I also went through some of the Mender Professional references in the
docs and added links to hosted mender where it seemed reasonable.

changelog: none

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

### CODE REVIEW

File: 04.Artifacts/05.Modifying-a-Mender-Artifact/docs.md [hunk 1]
```diff
@@ -37,9 +37,9 @@ MENDER_TENANT_TOKEN='<YOUR-MENDER-TENANT-TOKEN>'
 mender-artifact modify artifact.mender --server-uri "$MENDER_SERVER_URL" --tenant-token "$MENDER_TENANT_TOKEN"
 ```
 
-!!! The `--tenant-token` parameter is only needed for multi-tenant Mender servers like Mender Professional. In Mender Professional you can find it under [My organization](https://hosted.mender.io/ui/?target=_blank#/settings/my-organization).
+!!! The `--tenant-token` parameter is only needed for multi-tenant Mender servers like [Hosted Mender](https://hosted.mender.io). For Hosted Mender you can find it under [My organization](https://hosted.mender.io/ui/?target=_blank#/settings/my-organization).
```
Comment (lluiscampos) [inline 04.Artifacts/05.Modifying-a-Mender-Artifact/docs.md]: I think this should still be Mender Professional in some way. For example:

"... multi-tenant Mender servers like Mender Professional or other Mender hosted offerings"

File: 05.Client-configuration/06.Installing/docs.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ input is desired.
 The setup is different depending on your server configuration and the most common cases
 are shown below. Use `mender setup --help` to learn about all configuration options.
 
-- Connecting to Mender Professional with demo settings
+- Connecting to [Hosted Mender](https://hosted.mender.io) using demo settings
```
Comment (lluiscampos) [inline 05.Client-configuration/06.Installing/docs.md]: I would keep Mender Professional here too. And also, I think the link should go to https://mender.io/products/mender-professional

File: 04.Artifacts/05.Modifying-a-Mender-Artifact/docs.md (hunk 1 above)
Comment (alfrunes) [reply 04.Artifacts/05.Modifying-a-Mender-Artifact/docs.md]: I'm not sure about "other Mender hosted offerings", but I agree, I should probably revert to Mender professional (and perhaps mention enterprise) and provide links to the product page(s).

File: 05.Client-configuration/06.Installing/docs.md (hunk 2 above)
Comment (alfrunes) [reply 05.Client-configuration/06.Installing/docs.md]: In this case I disagree. If I was a user and clicking on a link that didn't take me to a login page would make me confused. I could probably replace `Hosted Mender` with `[hosted.mender.io](https://hosted.mender.io)` if that is less confusing.

**Comment by estenberg** [comment]
Sorry, too late for this one but this looks ok.

In general we should not be afraid of saying "hosted Mender" I think -- just like we are saying "onpremise Mender". The only change is that we added some plans on top of it which are called "Professional" (and we should rarely refer to the plans themselves, only if we ask people to sign up or it is a prerequisite for following a tutorial).

**Pull request merged**